Core/Core/Courses/CourseListView.swift

```diff
@@ -26,7 +26,16 @@ public struct CourseListView: View {
     @Environment(\.appEnvironment) var env: AppEnvironment
     @Environment(\.viewController) var controller: () -> UIViewController?
     @ObservedObject var allCourses: Store<GetAllCourses>
-    @State private var filter = ""
+
+    public class Props: ObservableObject {
+        @Published public var filter: String
```

Comment:
Was this because state was hard to test?

Reply:
Yes, although I didn't try testing State again this round